Android compile error -- 'pthread_condattr_setclock' undefined in 'Event_POSIX.cpp'

Using Poco v1.7.8 with Android NDK r15c (building for API level 19), I get the following error compiling 'Event_POSIX.cpp'.

`Foundation/Event_POSIX.cpp:72:6: error: use of undeclared identifier 'pthread_condattr_setclock'; did you mea 'pthread_condattr_setpshared'? if (pthread_condattr_setclock(&attr, CLOCK_MONOTONIC)) ^~~~~~~~~~~~~~~~~~~~~~~~~ pthread_condattr_setpshared Android/NDK/build//../sysroot/usr/include\pthread.h:128:5: note: 'pthread_condattr_setpshared' declared here int pthread_condattr_setpshared(pthread_condattr_t* _Nonnull, int); ^ 1 error generated.
`
This appears to be similar to GH [#1403](https://github.com/pocoproject/poco/issues/1403), which was labeled as "fixed" by v1.7.8.  Did something change in NDK r15c to cause the issue to resurface?

If I change my API level to 21, the problem of course goes away.  It also goes away if I switch back to NDK v13, though it's not as clear (to me) why.

Is API level 19 using these versions of Poco & NDK supported?

--- a/Foundation/src/Event_POSIX.cpp
+++ b/Foundation/src/Event_POSIX.cpp
@@ -24,12 +24,12 @@
 
 //
 // Note: pthread_cond_timedwait() with CLOCK_MONOTONIC is supported
-// on Linux and QNX, as well as on Android >= 5.0. On Android < 5.0,
-// HAVE_PTHREAD_COND_TIMEDWAIT_MONOTONIC is defined to indicate
-// availability of non-standard pthread_cond_timedwait_monotonic().
+// on Linux and QNX, as well as on Android >= 5.0 (API level 21).
+// On Android < 5.0, HAVE_PTHREAD_COND_TIMEDWAIT_MONOTONIC is defined
+// to indicate availability of non-standard pthread_cond_timedwait_monotonic().
 //
 #ifndef POCO_HAVE_MONOTONIC_PTHREAD_COND_TIMEDWAIT
-	#if (defined(__linux__) || defined(__QNX__)) && !(defined(__ANDROID__) && defined(HAVE_PTHREAD_COND_TIMEDWAIT_MONOTONIC))
+	#if (defined(__linux__) || defined(__QNX__)) && !(defined(__ANDROID__) && (defined(HAVE_PTHREAD_COND_TIMEDWAIT_MONOTONIC) || __ANDROID_API__ <= 21))
 		#define POCO_HAVE_MONOTONIC_PTHREAD_COND_TIMEDWAIT 1
 	#endif
 #endif

--- a/Foundation/src/Semaphore_POSIX.cpp
+++ b/Foundation/src/Semaphore_POSIX.cpp
@@ -24,12 +24,12 @@
 
 //
 // Note: pthread_cond_timedwait() with CLOCK_MONOTONIC is supported
-// on Linux and QNX, as well as on Android >= 5.0. On Android < 5.0,
-// HAVE_PTHREAD_COND_TIMEDWAIT_MONOTONIC is defined to indicate
-// availability of non-standard pthread_cond_timedwait_monotonic().
+// on Linux and QNX, as well as on Android >= 5.0 (API level 21).
+// On Android < 5.0, HAVE_PTHREAD_COND_TIMEDWAIT_MONOTONIC is defined
+// to indicate availability of non-standard pthread_cond_timedwait_monotonic().
 //
 #ifndef POCO_HAVE_MONOTONIC_PTHREAD_COND_TIMEDWAIT
-	#if (defined(__linux__) || defined(__QNX__)) && !(defined(__ANDROID__) && defined(HAVE_PTHREAD_COND_TIMEDWAIT_MONOTONIC))
+	#if (defined(__linux__) || defined(__QNX__)) && !(defined(__ANDROID__) && (defined(HAVE_PTHREAD_COND_TIMEDWAIT_MONOTONIC) || __ANDROID_API__ <= 21))
 		#define POCO_HAVE_MONOTONIC_PTHREAD_COND_TIMEDWAIT 1
 	#endif
 #endif

